packages/babel-jest/src/index.js

```diff
@@ -36,20 +35,24 @@ const createTransformer = (options: any): Transformer => {
     plugins: (options && options.plugins) || [],
     presets: ((options && options.presets) || []).concat(jestPresetPath),
     sourceMaps: 'both',
-  });
+  };
 
   delete options.cacheDirectory;
   delete options.filename;
 
+  const loadBabelConfig = (cwd, filename) =>
+    // `cwd` first to allow incoming options to override it
+    loadPartialConfig({cwd, ...options, filename});
+
   return {
     canInstrument: true,
     getCacheKey(
       fileData: string,
       filename: Path,
       configString: string,
-      {instrument, rootDir}: CacheKeyOptions,
+      {config, instrument, rootDir}: {config: ProjectConfig} & CacheKeyOptions,
```

Comment:
I think `CacheKeyOptions` already has the `config` property https://github.com/facebook/jest/blob/94c40a5dc5df86b8f27e403bde7fff1e33c74e16/types/Transform.js#L29